src/wpt/harness.ts

```diff
@@ -66,8 +68,208 @@ type TestCase = {
+      this_obj = this;
+    }
+
+    try {
+      return func.call(this_obj, ...rest);
+    } catch (err) {
+      if (this.phase >= TestPhase.HAS_RESULT) {
+        return undefined;
+      }
+
+      this.error = new AggregateError([err], this.name);
+      this.done();
+    }
+
+    return undefined;
+  }
+
+  /**
+   * Wrap a function so that it runs as a step of the current test.
+   *
+   * This allows creating a callback function that will run as a
+   * test step.
+   *
+   * @example
+   * let t = async_test("Example");
+   * onload = t.step_func(e => {
+   *   assert_equals(e.name, "load");
+   *   // Mark the test as complete.
+   *   t.done();
+   * })
+   *
+   * @param {string} func - Function to run as a step. If this
+   * throws an :js:func:`AssertionError`, or any other exception,
+   * the :js:class:`Test` status is set to ``FAIL``.
+   * @param {Object} [this_obj] - The object to use as the this
+   * value when calling ``func``. Defaults to the :js:class:`Test` object.
+   */
+  public step_func(func: UnknownFunc, this_obj?: object): UnknownFunc {
+    // eslint-disable-next-line @typescript-eslint/no-this-alias -- We could get rid of this this alias but the code is copied from WPT
+    const test_this = this;
+
+    if (arguments.length === 1) {
+      // eslint-disable-next-line @typescript-eslint/no-this-alias -- WPT allows for overriding the this environment for a step but defaults to the Test class
+      this_obj = this;
+    }
+
+    return function (...params: unknown[]) {
+      return test_this.step.call(test_this, func, this_obj, ...params);
+    };
+  }
+
+  /**
+   * Wrap a function so that it runs as a step of the current test,
+   * and automatically marks the test as complete if the function
+   * returns without error.
+   *
+   * @param {string} func - Function to run as a step. If this
+   * throws an :js:func:`AssertionError`, or any other exception,
+   * the :js:class:`Test` status is set to ``FAIL``. If it returns
+   * without error the status is set to ``PASS``.
+   * @param {Object} [this_obj] - The object to use as the this
+   * value when calling `func`. Defaults to the :js:class:`Test` object.
+   */
+  public step_func_done(func?: UnknownFunc, this_obj?: object): UnknownFunc {
+    // eslint-disable-next-line @typescript-eslint/no-this-alias -- WPT allows for overriding the this environment for a step but defaults to the Test class
+    const test_this = this;
+
+    if (arguments.length === 1) {
+      this_obj = test_this;
+    }
+
+    return function (...params: unknown[]) {
+      if (func) {
+        test_this.step.call(test_this, func, this_obj, ...params);
+      }
+
+      test_this.done();
+    };
+  }
+
+  /**
+   * Return a function that automatically sets the current test to
+   * ``FAIL`` if it's called.
+   *
+   * @param {string} [description] - Error message to add to assert
+   * in case of failure.
+   *
+   */
+  public unreached_func(description: string): UnknownFunc {
```

Comment:
this should be optional. in jsdoc, surrounding the name of the variable with square brackets make it optional. ex: [description]

Ref: https://www.typescriptlang.org/docs/handbook/jsdoc-supported-types.html#param-and-returns

```js
public unreached_func(description?: string): UnknownFunc {
```